src/Dialog/index.tsx

```diff
@@ -121,8 +121,8 @@ const Dialog = ({
   // the shim has events for mouse users only; does not require a role
   /* eslint-disable jsx-a11y/no-static-element-interactions,jsx-a11y/click-events-have-key-events */
   const dialogJSX = (
-    <div className="nds-dialog-root">
-      <CSSTransition timeout={1} classNames="nds-dialog-transition" appear in>
+    <CSSTransition timeout={1} classNames="nds-dialog-transition" appear in>
```

Comment:
I moved the CSSTransition to the outermost element here to restore the transition. It actually transitions now.